### PR TITLE
Implement versioning scheme based on ldflags

### PR DESCRIPTION
**Pull request by aguerra**
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa-api/221)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,8 +66,11 @@ shell:
 run-api-server:
 	go run ./cmd/server/main.go run --port 8080
 
+build-server:
+	@go build -ldflags "-X $(TERESA_API)/pkg/version.Version=$(TERESA_VERSION)" -o teresa-server $(TERESA_API)/cmd/server
+
 build-client:
-	@go build -o teresa cmd/client/main.go
+	@go build -ldflags "-X $(TERESA_API)/pkg/version.Version=$(TERESA_VERSION)" -o teresa $(TERESA_API)/cmd/client
```
Comment (drgarcia1986) [inline Makefile]: :clap:

File: Makefile [hunk 2]
```diff
@@ -64,8 +66,11 @@ shell:
 run-api-server:
 	go run ./cmd/server/main.go run --port 8080
 
+build-server:
```
Comment (drgarcia1986) [inline Makefile]: There are a few differences in `build-server` command between Makefile and Dockerfile, what do you think about unify?

File: Makefile [hunk 3]
```diff
@@ -3,6 +3,8 @@ IMAGE_NAME=teresa
 IMAGE_VERSION=0.1.1
 IMAGE_INSTANCE=default
 TERESA_DOCKER_PORT ?= 8080
+TERESA_VERSION ?= latest
```
Comment (drgarcia1986) [inline Makefile]: We can use current commit hash, will be awesome 😄

File: Makefile (hunk 2 above)
Comment (aguerra) [inline Makefile]: Sure, I was thinking of doing last.

**Pull request closed without merging**